### DISCUSSION TITLE
Yasim to JSBSim conversion, is it possible?

### DISCUSSION THREAD

**Discussion by demonixis** [JSBSim-Team/jsbsim#878] [Q&A]
Hello everyone,
I see a lot of Yasim models, especially helicopters. Is there a way to convert from Yasim to JSBSim? I've found [this python script](https://github.com/jan-hudec/yaamatic) that tries to do the job, but I don't understand how to use it.
If you've some resources to try this out thanks!

**Comment by ermarch** [comment]
I once lingered the idea to add support for YASim configuration files to AeromatiC++
But I've never looked at it to se how feasible that would be.

**Reply by demonixis** [reply]
So is that possible? I mean it could be great because a LOT of models in flightgear are using YASIM. So having this feature can help a lot 3rd party sims to supports more aircrafts, especially helicopters (this is why I ask)

**Reply by ermarch** [reply]
In theory it would be possible, but as I said I haven't tried anything yet.

Helicopters is a whole different matter. JSBSim helicopter support eludes me, and probably most other developers too.

So AeromatiC++ does not support creating helicopters.

**Reply by demonixis** [reply]
OK so if AeromaticC++ can convert from Yasim to Jsbsim, is it probably a few changes to make it compatible with helicopters?

**Reply by ermarch** [reply]
Not really. Both concepts are so different that converting regular aircraft from YASim to JSBSim using AeromatiC++ is not straightforward.
And doing the same for helicopters adds a whole new level of complexity.

**Reply by demonixis** [reply]
ok I understand. Do you know if it's possible to compile Yasim as a library? To use it with Unreal for instance?

**Reply by seanmcleod** [reply]
Given that FlightGear can toggle between aircraft using different FDM engines, i.e. JSBSim or Yasim I'd be surprised if Yasim isn't also built as a library.

**Comment by ermarch** [comment]
After taking a quick look it doesn't look like YASIm us build as a library.
But that doesn't mean it can't be don of course.

That said, there is no license information in the source files which means you will need to assume it's that of the main project (FlightGear) which is GPLv2+ and not LGPLv2+ like JSBSim

Erik
